windows/RNFetchBlob/RNFetchBlob.cpp

```diff
@@ -0,0 +1,1642 @@
+		promise.Reject(winrt::to_string(ex.message()).c_str());
+	}
+}
+
+
+// ls
+winrt::fire_and_forget RNFetchBlob::ls(
+	std::string path,
+	winrt::Microsoft::ReactNative::ReactPromise<std::vector<std::string>> promise) noexcept
+	try
+{
+	winrt::hstring directoryPath, fileName;
+	splitPath(path, directoryPath, fileName);
+
+	StorageFolder targetDirectory{ co_await StorageFolder::GetFolderFromPathAsync(directoryPath) };
+
+	std::vector<std::string> results;
+	auto items{ co_await targetDirectory.GetItemsAsync() };
+	for (auto item : items)
+	{
+		results.push_back(to_string(item.Name()));
+	}
+	promise.Resolve(results);
+}
+catch (const hresult_error& ex)
+{
+	hresult result{ ex.code() };
+	if (result == 0x80070002) // FileNotFoundException
+	{
+		promise.Reject(winrt::Microsoft::ReactNative::ReactError{ "ENOTDIR", "Not a directory '" + path + "'" });
+	}
+	else
+	{
+		promise.Reject(winrt::Microsoft::ReactNative::ReactError{ "EUNSPECIFIED", winrt::to_string(ex.message()).c_str() });
+	}
+}
+
+
+// mv
+winrt::fire_and_forget RNFetchBlob::mv(
+	std::string src, // from
+	std::string dest, // to
+	std::function<void(std::string)> callback) noexcept
+{
+	try
+	{
+		winrt::hstring srcDirectoryPath, srcFileName;
+		splitPath(src, srcDirectoryPath, srcFileName);
+
+		winrt::hstring destDirectoryPath, destFileName;
+		splitPath(dest, destDirectoryPath, destFileName);
+
+		StorageFolder srcFolder{ co_await StorageFolder::GetFolderFromPathAsync(srcDirectoryPath) };
+		StorageFolder destFolder{ co_await StorageFolder::GetFolderFromPathAsync(destDirectoryPath) };
+		StorageFile file{ co_await srcFolder.GetFileAsync(srcFileName) };
+
+		co_await file.MoveAsync(destFolder, destFileName, NameCollisionOption::ReplaceExisting);
+		callback("");
+	}
+	catch (const hresult_error& ex)
+	{
+		hresult result{ ex.code() };
+		if (result == 0x80070002) // FileNotFoundException
+		{
+			callback("Source file not found.");
+		}
+		else
+		{
+			callback(winrt::to_string(ex.message()).c_str());
+		}
+	}
+}
+
+
+// cp
+winrt::fire_and_forget RNFetchBlob::cp(
+	std::string src, // from
+	std::string dest, // to
+	std::function<void(std::string)> callback) noexcept
+	try
+{
+	winrt::hstring srcDirectoryPath, srcFileName;
+	splitPath(src, srcDirectoryPath, srcFileName);
+
+	winrt::hstring destDirectoryPath, destFileName;
+	splitPath(dest, destDirectoryPath, destFileName);
+
+	StorageFolder srcFolder{ co_await StorageFolder::GetFolderFromPathAsync(srcDirectoryPath) };
+	StorageFolder destFolder{ co_await StorageFolder::GetFolderFromPathAsync(destDirectoryPath) };
+	StorageFile file{ co_await srcFolder.GetFileAsync(srcFileName) };
+
+	co_await file.CopyAsync(destFolder, destFileName, NameCollisionOption::FailIfExists);
+
+	callback("");
+}
+catch (const hresult_error& ex)
+{
+	hresult result{ ex.code() };
+	if (result == 0x80070002) // FileNotFoundException
+	{
+		callback("Source file not found.");
+	}
+	callback(winrt::to_string(ex.message()).c_str());
```

Comment:
When the error is 0x8007002, `callback` is called twice, which is causing a runtime error crash.